### PR TITLE
Hmmbase

### PR DESCRIPTION
**Pull request by mortonjt**
This creates the abstract base class and the child classes for wrapping Infernal and HMMER

### CODE REVIEW

File: micronota/bfillings/hmmer.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        i: _get_parameter(FlagParameter, i)
+        for i in _flag_options})
+
+
+class HMMFetch(ModelFetch):
+    '''hmmfetch application controller.
+
+    hmmfetch is used to get a covariance model by name or accession
+    from a HMM database.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014).
```
Comment (RNAer) [inline micronota/bfillings/hmmer.py]: change INFERNAL to HMMER

File: micronota/bfillings/hmmer.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    '''
+    pass
+
+
+class HMMPress(ModelPress):
+    '''hmmpress application controller.
+
+    hmmpress is used to format a HMM database into a binary format for hmmscan.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014).
```
Comment (RNAer) [inline micronota/bfillings/hmmer.py]: Same here

File: micronota/bfillings/infernal.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+from .model import ModelFetch, ModelPress, ModelScan
+
+class CMScan(ModelScan):
+    '''cmscan application controller.
+
+    cmscan is used to search a sequence against a covariance model database.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014)
+    '''
+    _command = "cmscan"
+    _valued_path_options = ModelScan._valued_path_options
```
Comment (RNAer) [inline micronota/bfillings/infernal.py]: this line is not necessary. it is inherited by default.

File: .travis.yml [hunk 4]
```diff
@@ -31,4 +31,4 @@ notifications:
     urls:
       - https://webhooks.gitter.im/e/730436c08fd7f89462b0
     on_success: change
-    on_failure: always
+    on_failure: always
```
Comment (RNAer) [inline .travis.yml]: what is this?

File: micronota/bfillings/hmmer.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+class HMMScan(ModelScan):
+    '''hmmscan application controller.
+
+    hmmscan is used to search a sequence against a covariance model database.
+    This wrapper is tested for HMMER 3.1b1 (May 2013)
+    '''
+    _command = "hmmscan"
+
+    _valued_path_options = ModelScan._valued_path_options
```
Comment (RNAer) [inline micronota/bfillings/hmmer.py]: this line is not necessary. it is inherited by default.

File: micronota/bfillings/tests/data/hmmer/Pfam_B_1.fasta [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+RESARRDKTLLKEKRK
+>E7DZT2_9SAUR/1-69 PB000001; Pfam-B_1;
+PMKEISIRSXGSLAYPGLRTQGTLENPGGPNDASRGLTSLADTFEHVIEELLDEQQDIQPSKENKEADL
+>F1BZW2_CALMI/1-41 PB000001; Pfam-B_1;
+MTILFLTMVISYFGCLKAAPMKEAGGRGQGGSLYPALRPHG
+>F6JXV7_TRASC/9-96 PB000001; Pfam-B_1;
+MTILFLTMVISYFSCMKAAPMKEASVRGLGSLAYPGLRTHGTLESLSGPNTGSRGLTSLADTFEHVIEELLDEEQDVQPS
+EENKDADL
+>F6QKQ6_ORNAN/1-109 PB000001; Pfam-B_1;
+MHRLSFKNLKSTAENLWNSHFIRLRGSIKNGSRARSPNPTSGLSEGDEGRPSQPLPRNQEL
```
Comment (RNAer) [inline micronota/bfillings/tests/data/hmmer/Pfam_B_1.fasta]: add a newline at the end of the file?

File: micronota/bfillings/infernal.py (hunk 3 above)
Comment (mortonjt) [inline micronota/bfillings/infernal.py]: Ok, this is where things get fuzzy.

So _value_path_options is not actually an instance variable - it is a class variable.
Class variables as it turns out are not implicitly inherited - they need to be explicitly inherited.

Just as a sanity check, I ran this by commenting out `_valued_path_options`, and it fails as expected

File: micronota/bfillings/tests/data/hmmer/Pfam_B_1.fasta (hunk 6 above)
Comment (mortonjt) [inline micronota/bfillings/tests/data/hmmer/Pfam_B_1.fasta]: :L

File: .travis.yml (hunk 4 above)
Comment (mortonjt) [inline .travis.yml]: An accident waiting to be corrected.

File: micronota/bfillings/tests/test_hmmer.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
+from tempfile import mkstemp
+from os import getcwd, remove, close
+from unittest import TestCase, main
+from functools import partial
+from skbio.util import get_data_path
+from burrito.util import ApplicationError
+from os.path import join
+from micronota.bfillings.hmmer import (HMMScan, hmmscan_fasta,
+                                       hmmpress_hmm)
```
Comment (RNAer) [inline micronota/bfillings/tests/test_hmmer.py]: the import should be ordered by builtin module, dependency module, and module of this pkg (separated by a empty line between the three).

File: micronota/bfillings/tests/test_hmmer.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+
+class HMMERTests(TestCase):
+
```
Comment (RNAer) [inline micronota/bfillings/tests/test_hmmer.py]: does this empty line cause pep8 warning? do you run `make test`?

**Comment by RNAer** [comment]
any luck? @mortonjt

**Comment by mortonjt** [comment]
That should fix it

**Comment by RNAer** [comment]
The only thing I did was to restart the Travis build. Now it is passed! It seems it was the problem of travis.

**Comment by mortonjt** [comment]
Sweet

**Pull request merged**